lib/src/Base/Solver/LeastSquaresEquationsSolver.cxx

```diff
@@ -0,0 +1,144 @@
+  : SolverImplementation(absoluteError, relativeError, residualError, maximumCallsNumber)
+{
+  // LeastSquares problem
+  LeastSquaresProblem problem(SymbolicFunction("x", "x"));
+  solver_ = OptimizationAlgorithm::Build(problem);
+  solver_.setMaximumCallsNumber(maximumCallsNumber);
+  solver_.setMaximumAbsoluteError(absoluteError);
+  solver_.setMaximumRelativeError(relativeError);
+  solver_.setMaximumResidualError(residualError);
+}
+
+/* Virtual constructor */
+LeastSquaresEquationsSolver * LeastSquaresEquationsSolver::clone() const
+{
+  return new LeastSquaresEquationsSolver(*this);
+}
+
+/* String converter */
+String LeastSquaresEquationsSolver::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << LeastSquaresEquationsSolver::GetClassName()
+      << " derived from " << SolverImplementation::__repr__();
+  return oss;
+}
+
+void LeastSquaresEquationsSolver::setOptimizationAlgorithm(const OptimizationAlgorithm & algorithm)
+{
+  solver_ = algorithm;
+}
+
+OptimizationAlgorithm LeastSquaresEquationsSolver::getOptimizationAlgorithm() const
+{
+  return solver_;
+}
+
+/* Solve attempt to find one root to the system of non-linear equations function(x) = 0
+   given a starting point x with a least square optimization method.
+*/
+Point LeastSquaresEquationsSolver::solve(const Function & function,
+                                         const Point & startingPoint) const
+{ 
+  const Interval bounds;
+  return solve(function, startingPoint, bounds);
+}
+
+/* Solve attempt to find one root to the system of non-linear equations function(x) = 0
+   given a starting point x with a least square optimization method.
+*/
+Point LeastSquaresEquationsSolver::solve(const Function & function,
+                                         const Point & startingPoint,
+                                         const Interval & bounds) const
+{
+  UnsignedInteger callsNumber = 0;
+  LeastSquaresProblem lsqProblem(function);
+  const UnsignedInteger boundsDimension = bounds.getDimension();
+  if (boundsDimension == function.getInputDimension())
+    lsqProblem.setBounds(bounds);
+  if ((boundsDimension > 0) && (boundsDimension != function.getInputDimension()))
+      throw InvalidArgumentException(HERE) << "Bounds should be of dimension 0 or dimension = " << function.getInputDimension()
+                                           << ". Here bounds's dimension = " << boundsDimension;
+  OptimizationAlgorithm solver(solver_);
+  solver.setStartingPoint(startingPoint);
+  try
+  {
+    solver.setProblem(lsqProblem);
+  }
+  catch (const InvalidArgumentException &)
+  {
+    LOGWARN("Default optimization algorithm could not solve the least squares problem. Trying to set up a new one...");
+    solver = OptimizationAlgorithm::Build(lsqProblem);
+  }
+  solver.setProblem(lsqProblem);
+  solver.run();
+  callsNumber = solver.getResult().getCallsNumber();
+  callsNumber_ = callsNumber;
```

Comment:
seems the callsNumber variable is not very useful

Reply:
yes